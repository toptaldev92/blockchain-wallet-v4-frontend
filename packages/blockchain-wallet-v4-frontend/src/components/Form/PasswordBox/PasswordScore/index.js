import React from 'react'

import { PasswordGauge } from 'blockchain-info-components'

const PasswordScore = props => {
  return <PasswordGauge score={props.passwordScore} />
}

export default PasswordScore
