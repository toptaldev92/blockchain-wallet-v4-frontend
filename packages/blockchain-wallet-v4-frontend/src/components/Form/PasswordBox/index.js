import PropTypes from 'prop-types'
import React from 'react'
import styled from 'styled-components'

import {
  Icon,
  PasswordGauge,
  PasswordInput,
  Text
} from 'blockchain-info-components'

const Container = styled.div`
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: flex-start;
  width: 100%;
  border-radius: 8px;
  overflow: hidden;
`
const Error = styled(Text)`
  position: absolute;
  display: block;
  height: 15px;
  top: -20px;
  right: 0;
`

const WarningIcon = styled(Icon)`
  position: absolute;
  margin: auto 0;
  right: 16px;
  top: 14px;
`

const getErrorState = ({ touched, invalid }) => {
  return touched && invalid ? 'invalid' : 'initial'
}

const PasswordBox = field => {
  const {
    meta,
    input,
    showPasswordScore,
    disabled,
    borderColor,
    noLastPass,
    passwordScore
  } = field
  const { touched, error, active } = meta
  const errorState = getErrorState(meta)
  const scoreVisible = showPasswordScore ? input.value.length > 0 : false

  return (
    <Container>
      <PasswordInput
        {...input}
        disabled={disabled}
        active={active}
        controlledBorderColor={borderColor}
        errorState={errorState}
        data-e2e={field['data-e2e']}
        noLastPass={noLastPass}
      />
      {scoreVisible && <PasswordGauge score={passwordScore + 1} />}
      {touched && error && (
        <>
          <Error
            size='12px'
            weight={500}
            color='error'
            data-e2e='passwordsNotMatchError'
          >
            {error}
          </Error>
          {noLastPass && (
            <WarningIcon name='alert-filled' color='red600' size='20px' />
          )}
        </>
      )}
    </Container>
  )
}

PasswordBox.propTypes = {
  score: PropTypes.bool
}

PasswordBox.defaultProps = {
  score: false
}

export default PasswordBox
