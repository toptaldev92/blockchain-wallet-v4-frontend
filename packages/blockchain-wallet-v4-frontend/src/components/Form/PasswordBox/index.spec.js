import { shallow } from 'enzyme'
import PasswordBox from './index.js'
import React from 'react'
import toJson from 'enzyme-to-json'

jest.mock('blockchain-info-components', () => {
  return {
    Text: jest.fn(),
    Icon: jest.fn(),
    PasswordInput: jest.fn()
  }
})

describe('PasswordBox', () => {
  it('renders correctly', () => {
    const props = { meta: {} }
    const component = shallow(<PasswordBox {...props} />)
    const tree = toJson(component)
    expect(tree).toMatchSnapshot()
  })
})
