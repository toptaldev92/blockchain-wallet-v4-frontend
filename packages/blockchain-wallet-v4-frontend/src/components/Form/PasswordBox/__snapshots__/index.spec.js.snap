// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordBox renders correctly 1`] = `
<styled.div>
  <mockConstructor
    errorState="initial"
  />
</styled.div>
`;
